### ISSUE TITLE
SB spies for addon-actions

### ISSUE THREAD

**Issue by shilman** [storybookjs/storybook#15517]
As a user, I'd like to be able to assert that actions have been called so that I can verify event behavior in my components.

We should add hooks to addon-actions to automatically make them [spies](https://github.com/facebook/jest/tree/master/packages/jest-mock), so we get `expect(xx).toHaveBeenCalled` for free.

Related to #15515

**Comment by yannbf** [comment]
> Does `addon-interactions.stories.tsx` still work if you remove this:
> 
> ```
>   argTypes: {
>     onSubmit: { action: true },
>   },
> ```

Yes it does! As long as you set the `argTypeRegex` in the `actions` parameter like so:
```
export const parameters = { 
    actions: { argTypesRegex: '^on.\*' } 
}
```

Closing the issue as the functionality has been implemented already.
